proposals/extended-property-patterns.md

```diff
@@ -0,0 +1,44 @@
+# Extended property patterns
+
+## Summary
+[summary]: #summary
+
+Allow property subpatterns to reference nested members.
```

Comment:
Let's add an example to illustrate up front.

Also, let's add a link to property patterns spec somewhere in this doc: [https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/patterns.md#property-pattern](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-8.0/patterns.md#property-pattern)
Take example on it to write this one.

Reply:
Didn't find anything in the proposal to include. 

Added an example that I encountered first that was kinda annoying :)